docs/resources/Channel.md

```diff
@@ -27,6 +28,8 @@ Represents a guild or DM channel within Discord.
 | parent_id?             | ?snowflake                                                             | id of the parent category for a channel (each parent category can contain up to 50 channels)                                                                                    |
 | last_pin_timestamp?    | ISO8601 timestamp                                                      | when the last pinned message was pinned                                                                                                                                         |
 
+\* The `name` field is only nullable for group DM channels.
```

Comment:
This isn't a complaint at this PR, but a question for the next employee who reads this, if it is read. 

At this point, would it be more helpful to have separate tables for each type of channel? As a developer, this kind of merged information makes writing distinct entities for each type of channel incredibly awkward and a case of trial and error. Since general programming paradigms encourage separation of concerns, that would likely be of much more use. Especially since the majority of programming languages interfacing with this API for anything other than a trivial request will likely want to make some form of struct or class to represent the entity they are dealing with.

That would remove the need to document stuff like this in this way as well, and remove questions such as "can you pin messages in a news channel? Is a news channel the same as a text channel, because I cant tell because I am not lucky enough to be the owner of a guild that has a news channel to try this on? Can voice channels really be marked as NSFW?".

Reply:
I can agree that a separation could be beneficial towards having an easier to understand documentation.

Towards your questions;
- News channels are - in most basic terms - just fancy Text Channels with the option that you can post news to following Guilds. Most libraries I know just treat News Channels as normal Text Channels as they, from what I know, don't offer any specific functions for bots that would be different to normal Text Channels. So yes, you could pin messages in that channel as it would be a normal Text Channel with some slight differences.
- Voice Channels can't be marked nsfw. You probs can set the NSFW property (Since the docs suggest that it is a thing) but I believe that Discord just ignores that value for the Voice Channel type, so no matter what, it would remain false... I could be wrong here tho.

Reply:
Cheers for the clarification :)

Reply:
nsfw can be set for any guild channel, if it's being ignored that's at a client level not the api level.